x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/utils/MapExtractor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiler.utils;
+
+import java.util.Map;
+
+public final class MapExtractor {
+    private MapExtractor() {}
+
+    @SuppressWarnings("unchecked")
+    public static <T> T read(Map<String, Object> source, String... path) {
```

Comment:
I am assuming it is not possible to reuse ObjectPath?

Reply:
Thanks! I was not aware of that class and replaced all usages of `MapExtractor` now with `ObjectPath`.